pypeit/scripts/ql_mos.py

```diff
@@ -0,0 +1,110 @@
+#!/usr/bin/env python
+#
+# See top-level LICENSE file for Copyright information
+#
+# -*- coding: utf-8 -*-
+"""
+This script runs PypeIt on a set of MultiSlit images
+"""
+import argparse
+
+from pypeit import msgs
+
+import warnings
+
+def parser(options=None):
+
+    parser = argparse.ArgumentParser(description='Script to run PypeIt in QuickLook on a set of MOS files')
+    parser.add_argument('spectrograph', type=str, help='Name of spectograph, e.g. shane_kast_blue')
+    parser.add_argument('full_rawpath', type=str, help='Full path to the raw files')
+    parser.add_argument('arc', type=str, help='Arc frame filename')
+    parser.add_argument('flat', type=str, help='Flat frame filename')
+    parser.add_argument('science', type=str, help='Science frame filename')
+    parser.add_argument('-b', '--box_radius', type=float, help='Set the radius for the boxcar extraction (arcsec)')
+    parser.add_argument('-d', '--det', type=int, default=1, help='Detector number')
+    parser.add_argument("--ignore_headers", default=False, action="store_true",
+                        help="Ignore bad headers?")
+    parser.add_argument("--user_pixflat", type=str, help="Use a user-supplied pixel flat (e.g. keck_lris_blue)")
+
+    if options is None:
+        pargs = parser.parse_args()
+    else:
+        pargs = parser.parse_args(options)
+    return pargs
+
+
+def main(pargs):
+
+    import os
+    import numpy as np
+
+    from IPython import embed
+
+    from pypeit import pypeit
+    from pypeit import pypeitsetup
+    from pypeit.core import framematch
+
+    spec = pargs.spectrograph
+
+    # Config the run
+    cfg_lines = ['[rdx]']
+    cfg_lines += ['    spectrograph = {0}'.format(spec)]
+    cfg_lines += ['    redux_path = {0}_A'.format(os.path.join(os.getcwd(),spec))]
+    cfg_lines += ['    detnum = {0}'.format(pargs.det)]
+    if pargs.ignore_headers:
+        cfg_lines += ['    ignore_bad_headers = True']
+    cfg_lines += ['[calibrations]']
+    cfg_lines += ['    [[scienceframe]]']
+    cfg_lines += ['        [[process]]']
+    cfg_lines += ['              cr_reject = False']
+    if pargs.user_pixflat is not None:
+        cfg_lines += ['    [[flatfield]]']
+        cfg_lines += ['        frame = {0}'.format(pargs.user_pixflat)]
+    cfg_lines += ['[scienceimage]']
+    cfg_lines += ['    [[extraction]]']
+    cfg_lines += ['         boxcar_only = True']
+    if pargs.box_radius is not None: # Boxcar radius
+        cfg_lines += ['    boxcar_radius = {0}'.format(pargs.box_radius)]
+    cfg_lines += ['    [[findobj]]']
+    cfg_lines += ['         skip_second_find = True']
+
+    # Data files
+    data_files = [os.path.join(pargs.full_rawpath, pargs.arc),
+                  os.path.join(pargs.full_rawpath, pargs.flat),
+                  os.path.join(pargs.full_rawpath,pargs.science)]
+    # Damn JFH sorting
```

Comment:
I can't find it.  Please add it in.

Reply:
Not sure why are you sorting the data files since metadata does this sorting internally. Anyway, my fix to the sorting issue which was trivial, is line 1480 of write_sorted in metadata.py

Reply:
it's a bit more complicated than that.  Have dealt with it now.